src/submit_jobs.py

```diff
@@ -240,14 +258,23 @@ def read_job(job: SparkJob):
 
 if __name__ == "__main__":
     current_datetime = datetime.now().strftime("%Y-%m-%d_%H:%M:%S")
-    app_name = f"iasworld_{current_datetime}"
     default_args = load_yaml(PATH_DEFAULT_SETTINGS, "default_args")
     args = parse_args(default_args)
     logger.info(f"Starting Spark application with arguments: {args}")
+    app_name = f"iasworld_{args.extract_target}_{current_datetime}"
 
     try:
+        # Clear the existing extract files and logs before submitting new jobs.
+        # This is to prevent new jobs from becoming mixed with the results of
+        # previous failed or cancelled jobs
+        clear_directory(PATH_INITIAL_DIR)
+        clear_directory(PATH_FINAL_DIR)
+        if os.path.exists(PATH_SPARK_LOG):
+            os.remove(PATH_SPARK_LOG)
```

Comment:
We _really_ don't want to get the test and prod data mixed up, so I'm clearing the extract directory at the start of every run.